### PR TITLE
FEATURE: reconnect

### PR DESCRIPTION
**Pull request by Segelzwerg**
Requires

- [x] #356
- [x] #372

### CODE REVIEW

**Comment by Segelzwerg** [comment]
> So this doesnt do anything except for returning the room id?
> Why cant we use the player id to find their room?

If I remember correctly we are looking in the TableAtPlayer types which only knows the name and not the ID.

**Comment by Segelzwerg** [comment]
Shall we wait for #372 and then use the ID, @iTitus ?

File: tests/whist_server/base_token_case.py [hunk 1]
```diff
@@ -15,7 +16,8 @@
 
 class TestCaseWithToken(unittest.TestCase):
     def setUp(self) -> None:
-        self.player_mock = Player(username='marcel', rating=2000)
+        self.player_mock = UserInDb(id=ObjectId('1' * 24), username='marcel', rating=2000,
```
[the remark this answers is not in the document]
Comment (Segelzwerg) [reply tests/whist_server/base_token_case.py:19]: @sonatype-lift ignore false-positive: pydantic

**Comment by Segelzwerg** [comment]
@iTitus added status response. I think from RoomInfo + Stack everything can be reconstructed.

File: tests/whist_server/base_token_case.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Segelzwerg) [reply tests/whist_server/base_token_case.py:19]: @sonatype-lift ignore it is a test

[the remark this answers is not in the document]
Comment (Segelzwerg) [reply tests/whist_server/base_token_case.py:19]: @sonatype-lift ignore it is a test

File: whist_server/api/room/join.py [hunk 2]
```diff
@@ -81,7 +85,38 @@
...
+
+class ReconnectArguments(BaseModel):
+    """
+    Sets flags for extra response fields.
+    """
+    stack: bool = False
+
+
+@router.post('/reconnect/', status_code=200)
+def reconnect(args: ReconnectArguments = ReconnectArguments(stack=False),
```
[the remark this answers is not in the document]
Comment (Segelzwerg) [reply whist_server/api/room/join.py:103]: @sonatype-lift ignore it is BaseModel

**Comment by Segelzwerg** [comment]
@iTitus no it should be better. Can you review?

**Review by iTitus — APPROVED** [review]

**Pull request merged**